Add empty [workspace] to Cargo.toml.

mutexlox-signal opened the pull request:
This works around https://github.com/rust-lang/cargo/issues/6745 and allows cubeb-pulse-rs to appear as a subdirectory of a workspace member in another project (for instance, cubeb-rs).

mutexlox-signal committed:
Add empty [workspace] to Cargo.toml.

This works around https://github.com/rust-lang/cargo/issues/6745 and
allows cubeb-pulse-rs to appear as a subdirectory of a workspace
member in another project (for instance, cubeb-rs).

mutexlox-signal commented:
cc @padenot  - any chance of taking a look at this? thanks!

mutexlox-signal commented:
cc @kinetiknz on this one as well

mutexlox-signal commented:
Looks like this may have cause clippy to start running on things it wasn't running on before(?)

kinetiknz commented:
Yeah, clippy is now including pulse-rs and pulse-ffi.  Quick fix is to exclude those crates from the workspace (which would be equivalent to the behaviour before `[workspace]` was added), but it'd be better to fix the clippy errors.  Similar issue showing up in https://github.com/mozilla/cubeb-coreaudio-rs/pull/236 with coreaudio-sys-utils.

mutexlox-signal commented:
I think there are far too many errors for me to fix them in this PR, in particular as a number of them are due to missing `unsafe` tags, which may require invasive changes to address. For now, I will exclude these directories from the clippy check in question, and I have filed https://github.com/mozilla/cubeb-pulse-rs/issues/95 to be addressed as it fits in.

mutexlox-signal commented:
On second thought, I think the approach I took with that last commit is not *quite* right; it might also break anyone running `cargo clippy` manually or on projects that depend on this crate. Perhaps instead we should fix the 'easy' findings and silence the more involved ones.

mutexlox-signal committed:
Fix a bunch of clippy findings.

Largely autogenerated, with a bit of manual cleanup for extra
whitespace.

mutexlox-signal committed:
Manually fix some warnings and cargo fmt.

mutexlox-signal committed:
Silence a bunch of clippy warnings.

https://github.com/mozilla/cubeb-pulse-rs/issues/95 tracks fixing these
in the future.

mutexlox-signal commented:
@kinetiknz this is ready for a re-review now :)

( @padenot  - if you could please trigger a presubmit run so I can fix any lingering errors I missed locally, I'd appreciate it)

mutexlox-signal commented:
One of the nightly errors is due to https://github.com/rust-lang/rust-clippy/issues/13531. I can fix the others.

mutexlox-signal committed:
Disable broken clippy nightly check.

See https://github.com/rust-lang/rust-clippy/issues/13531

mutexlox-signal committed:
Fix some new clippy lifetime warnings.

mutexlox-signal commented:
And this should be ready for a re-run now, too

mutexlox-signal committed:
cargo fmt

mutexlox-signal commented:
reformatted

kinetiknz commented:
Thank you!

mutexlox-signal commented:
Thanks!
